msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CodesignTaskBase.cs

```diff
@@ -77,7 +77,9 @@ string GetFullPathToTool ()
 
 		string GetOutputPath (ITaskItem item)
 		{
-			return Path.Combine (StampPath, item.ItemSpec);
+			var path = item.ItemSpec;
+			var app = path.LastIndexOf (".app/");
```

Comment:
I'm not sure if it can actually happen, but this won't work with app extensions.

The original code used `AppBundleDir` to compute the relative path:

https://github.com/xamarin/xamarin-macios/blob/a5d06f0d1e9eeea7c34738677d5d78eca1582916/msbuild/Xamarin.iOS.Tasks.Core/Tasks/CodesignNativeLibrariesTaskBase.cs#L147

Reply:
My thoughts were that extensions, `.appex`, would be signed inside the `.app` of the container - giving them a unique , longer stamp name.

Otherwise you run into the risk of having the same native name (and path / stamp) inside the `.app` and `.appex` and since they could be different (like arch for watch) then the stamp would not a correct signature witness.

Reply:
> but this won't work with app extensions.

I double checked and what happens, right now, with extensions is:

```
    StampPath obj/Debug/codesign/
    outputPath obj/Debug/codesign/Debug/FinderSyncExtension.appex/Contents/MonoBundle/libmono-native.dylib
```

so the path is unique for each extension, so this should work. I'll compare with your suggestion...

Reply:
😞 
```
Tasks/CodesignTaskBase.cs(83,41): error CS0103: The name 'AppBundleDir' does not exist in the current context [/Users/poupou/git/xamarin/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Xamarin.MacDev.Tasks.Core.csproj]
```

Reply:
if I use
```csharp
string GetOutputPath (ITaskItem item)
{
	return Path.Combine (StampPath, PathUtils.AbsoluteToRelative (AppBundleDir, item.ItemSpec));
}
```

I get 
```
    /Users/poupou/Downloads/Xamarin_Mac___ExtensionSamples/FinderSyncExtension/bin/Debug/FinderSyncExtension.appex/Contents/MonoBundle/libmono-native.dylib: signed Mach-O thin (x86_64) [libmono-native-555549441d6f6137edfa3546a3fd036157175dae]
    
    StampPath obj/Debug/codesign/
    outputPath obj/Debug/codesign/Contents/MonoBundle/libmono-native.dylib
```

which is incorrect since it will overwrite the original (main app) and other (many extensions in this project) signature witness files